File: text/lord.pl
#! /usr/bin/perl

use v5.14;
use strict;

open(BIBLE, "<:utf8", "bible.txt");
binmode(STDOUT, 'utf:8');

my $counter = 0;
my $search = $ARGV[0];
my %appearences = ();


while (my $line = <BIBLE>)
{
    my @words = split(" ", $line);
    foreach my $word (@words)
    {
        my $lower = lc($word);

        unless ($appearences{$lower})
        {
            $appearences{$lower} = 0;
        }

        $appearences{$lower}++;
    }
}

say "The word $search appears $appearences{lc($search)} times on the King James Version of the Bible";

File: text/lord.c
/**
*
*       Find the occurencies of a ward in Bible (KJV)
*/

#include <stdio.h>
#include <hash.h>

int main(int argc, char** argv)
{
    if (argc != 2) return 1;
    HASH* map = mapf("bible.txt");

    printf("The word %s appears %li times in the Bible\n", argv[1], count(map, argv[1]));

    return 0;
}
